bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/Debouncer.java

```diff
@@ -74,7 +76,14 @@ void call() {
         lastCallAttempt = clock.millis();
         calls.incrementAndGet();
         if (pending.compareAndSet(false, true)) {
-            scheduler.schedule(this::tryActionOrPostpone, delayMs, TimeUnit.MILLISECONDS);
+            feature = scheduler.schedule(this::tryActionOrPostpone, delayMs, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    public void stop() {
+        logger.trace("stop debouncer");
+        if (feature != null) {
+            feature.cancel(true);
```

Comment:
should this also change your pending AtomicBoolean here to allow future call attempts?

Reply:
i was thinking of shutdown use cases, so that Debouncer instance will not get re-started anymore.. but probably it can be the case for bundle stop/start. so, i have added reset of pending and calls.